lib/matplotlib/backend_bases.py

```diff
@@ -3058,14 +3058,50 @@ def press_pan(self, event):
         if (event.button not in [MouseButton.LEFT, MouseButton.RIGHT]
                 or event.x is None or event.y is None):
             return
-        axes = [a for a in self.canvas.figure.get_axes()
-                if a.in_axes(event) and a.get_navigate() and a.can_pan()]
-        if not axes:
+
+        # get all relevant axes and group them by patch-visibility and zorder
+        all_axes = defaultdict(lambda: defaultdict(list))
+        [all_axes[a.patch.get_visible()][a.zorder].append(a)
+         for a in self.canvas.figure.get_axes()
+         if a.in_axes(event) and a.get_navigate() and a.can_pan()]
```

Comment:
Definitely do not use list comprehensions for their side effect.

Reply:
ok, fair point, using a normal loop now!